Cython/Compiler/MemoryView.py

```diff
@@ -841,17 +844,34 @@ def use_cython_array_utility_code(env):
     requires=[],  # require cython_array_utility_code
 )
 
-view_utility_code = load_memview_cy_utility(
+memoryview_utility_code = load_memview_cy_utility(
         "View.MemoryView",
         context=context,
-        requires=[Buffer.buffer_struct_declare_code,
+        requires=[
+                  Buffer.buffer_struct_declare_code,
+                  Buffer.buffer_formats_declare_code,
+                  memviewslice_init_code,
+                  is_contig_utility,
+                  overlapping_utility,
+                  copy_contents_new_utility,
+                  ],
+)
+
+shared_utility_code = load_memview_cy_shared_utility(
```

Comment:
Rather than always doing this, it should happen conditionally at need. The same applies to the `View.MemoryView` utility code above.